lib/reports/BackupReportManager.js

```diff
@@ -0,0 +1,224 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+const moment = require('moment-timezone');
+const Repository = require('../db').Repository;
+const CONST = require('../constants');
+const config = require('../config');
+
+class BackupReportManager {
+  static getInstanceBackupSummary(instance_id, start_time, end_time) {
+    return Promise
+      .all([
+        this.getBackupTriggerRecord(instance_id, start_time, end_time),
+        this.getBackupResult(instance_id, start_time, end_time),
+        this.getLastRunDetail(instance_id),
+        this.getJobDetails(instance_id)
+      ])
+      .spread((triggeredBackups, resultBackups, lastRunDetail, jobDetail) => {
+        const instance_record = {};
+        const noBackupDays = [];
+        //iterator to first entry of the day
+        let start_index = 0;
+        let threeConsecutiveBackupFailureCount = 0;
+        let currentDay = new Date(start_time);
+        let lastFailDay;
+        let secondlastFailDay;
+        let backupSuccessCount = 0;
+        if (jobDetail) {
+          if (start_time < moment.utc(jobDetail.createdAt).startOf('day').toDate()) {
+            currentDay = moment.utc(jobDetail.createdAt).startOf('day').toDate();
+          }
+          _.assign(instance_record, {
+            instanceCreateTime: jobDetail.createdAt
+          });
+        } else {
+          currentDay = new Date(end_time);
+        }
+        if (lastRunDetail && (end_time > lastRunDetail.createdAt)) {
+          _.assign(instance_record, {
+            instanceDeleteTime: lastRunDetail.createdAt
+          });
+          end_time = lastRunDetail.createdAt;
+        }
+        //iterate through list of days and for each day count successful backups
+        for (; currentDay < end_time; currentDay = moment(currentDay).add(1, 'days').toDate()) {
```

Comment:
currently unable to understand the logic from here till https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/113/files#diff-6e1b755c7de68f44c31e40d83c3897b8R71. 

Can you please provide some example ? Or maybe a UT will help.

Also better to extract this logic out into another function on which we can easily write an UT.